### PR TITLE
Kevinwulrich sensor subsystem

### PR DESCRIPTION
**Pull request by Kevinwulrich**

### CODE REVIEW

File: Documentation/Signoffs/sensor_subsystem.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+The goal of this subsystem is to accurately and quickly gather several data points to determine the
+distance, speed, and position of the target as it slides down the fishing line.
+
+## **Constraints:**
+
+| No. | Constraint                                                                                              | Origin            |
+| --- | ------------------------------------------------------------------------------------------------------- | ----------------- |
+| 1   | The sensor shall be supplied 5 V via USB from the Jetson Nano processor                                 | System Constraint |
+| 2   | The sensor shall be able to retrieve at least 2 data points within 1.95 s in order to calculate speed   | System Constraint |
+| 3   | The sensor shall have a resolution great enough so that the golf ball spans multiple pixels from 6 feet | Conceptual Design |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md]: How many is multiple? 2?

File: Documentation/Signoffs/sensor_subsystem.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+| 4   | The sensor shall have a range of at least 6 feet                                                        | Conceptual Design |
+| 5   | The sensor shall be able to detect and track a golf ball-sized object from a maximum of 6 feet          | System Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md]: If you say this, you will have to show me that the system can track the golf ball. Do you have a plan for analyzing that? Also, what does track mean?

File: Documentation/Signoffs/sensor_subsystem.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+| 6   | The sensor shall have a field of view greater than 24.50°                                               | Device Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md]: This constraint's reasoning does not suggest that this is correct. You aren't placing the sensor at the point of convergence. Therefore, the angle of the strings tells you little about this system in isolation.

File: Documentation/Signoffs/sensor_subsystem.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+The goal of this subsystem is to accurately and quickly gather several data points to determine the
+distance, speed, and position of the target as it slides down the fishing line.
+
+## **Constraints:**
+
+| No. | Constraint                                                                                              | Origin            |
+| --- | ------------------------------------------------------------------------------------------------------- | ----------------- |
+| 1   | The sensor shall be supplied 5 V via USB from the Jetson Nano processor                                 | System Constraint |
+| 2   | The sensor shall be able to retrieve at least 2 data points within 1.95 s in order to calculate speed   | System Constraint |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md]: This constraint does not allow you to actually have time to aim or decide when to fire. This is not a tight enough constraint.

File: Documentation/Signoffs/sensor_subsystem.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+| 3   | The sensor shall have a resolution great enough so that the golf ball spans 30 pixels from 6 feet       | Conceptual Design |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md]: Based on what?

Comment (JesseTNRoberts) [reply Documentation/Signoffs/sensor_subsystem.md]: How do you know that 30 is enough?

File: Documentation/Signoffs/sensor_subsystem.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+| 4   | The sensor shall have a range of at least 6 feet                                                        | Conceptual Design |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md]: What does this mean? Cameras don't typically have ranges?

File: Documentation/Signoffs/sensor_subsystem.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+~~~ math
+
+arctan((46.5 - 10) / (94 - 2)) = 21.64°
+
+~~~
+
+Using this angle will allow every line to be visible within the 42° vertical FOV of the RGB camera
+
+### **Resolution**
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/sensor_subsystem.md:45]: You have to explain what the math means with comments. Otherwise this is indecipherable.

**Review by micahr234 — APPROVED** [review]

**Pull request merged**